Code/Framework/AzCore/Tests/SystemFileTest.cpp

```diff
@@ -253,32 +253,54 @@ namespace UnitTest
 
     TEST_F(SystemFileTest, FileDescriptorCapturer_DoesNotDeadlock_WhenMoreThanPipeSizeContent_IsCaptured)
     {
-        AZStd::string stdoutData;
-        auto StoreStdout = [&stdoutData](AZStd::span<const AZStd::byte> capturedBytes)
+        using namespace AZ::IO;
+
+        AZStd::string capturedData;
```

Comment:
nit: Maybe `std::stringstream` is a more appropriate replacement? Its used in other O3DE test code.

Reply:
My thinking was it might be better that the sdtout replacement doesn't get involved with our memory tracking, but this is a small test, so i'm fine with it the way it is